### DISCUSSION TITLE
How should we use the new Doctrine Inflector rules?

### DISCUSSION THREAD

**Discussion by ninjasitm** [laravel/framework#33070] [General]
- Laravel Version: v6.18.16
- PHP Version: 7.3
- Database Driver & Version: Postgresql 12

### Description:

How should we upgrade the inflector to support custom pluralization rules given the new update in the Doctrin Inflector package? For reference see the Doctrine Inflector documentatio here: https://www.doctrine-project.org/projects/doctrine-inflector/en/2.0/index.html#custom-setup

Previous issues in this repo resolved the missing class such as: 

https://github.com/laravel/framework/issues/33043
https://github.com/laravel/framework/issues/32730

However there are no examples of how to use the updated library.

### Steps To Reproduce:
Previously this worked as a service:

```
<?php
namespace App\Providers;

use Illuminate\Support\Arr;
use Illuminate\Support\Str;
use Illuminate\Support\ServiceProvider;
use Doctrine\Inflector\Inflector;

class PluralizationServiceProvider extends ServiceProvider
{
    public function register()
    {
        Inflector::rules('plural', [
            'irregular' => [
                'octopus' => 'octopi',
                'coach' => 'coaches',
                'coaches' => 'coaches'
            ]
        ]);
    }
}
```

**Comment by rickselby** [comment]
Did you ever work out how to do this? I'm (finally) upgrading and have hit the same issue.
